### ISSUE TITLE
ApplePS2SmartTouchpad

### ISSUE THREAD

**Issue by wtdcode** [bavariancake/XPS9570-macOS#2]
Have you tried ApplePS2SmartTouchpad.kext to make the touchpad work well?

**Comment by wtdcode** [comment]
And does your touchscreen work?

**Comment by bavariancake** [comment]
I have not tried using ApplePS2SmartTouchpad, but if it works well for you, please send a pull request. 

The touchscreen does not work yet, because it's no longer a USB device (see notes about HS09 connector). It appears that Dell has switched to an I2C touchscreen, which may be supported with VoodoI2C. I haven't tried.

**Comment by wtdcode** [comment]
Thanks. 

My xps15 is still delivering and I will update this thread or send a pr if I have any further progress.

**Comment by eamigo86** [comment]
Hi @wtdcode, have you been able to get advanced gestures on the touchpad?

**Comment by eamigo86** [comment]
Hi please, read this: [Solve the problem of touch panel gesture and touch screen](https://github.com/Xigtun/xps-9570-mojave/issues/23)

**Comment by alexmenn** [comment]
Yes, if I use I2C from @eamigo86 post, then we get random kp on boot (sometimes I mean) and I cant boot without verbose mode. With normal voodoops2controller installed, it boots ok.

**Comment by d3myz** [comment]
Hi All, I have been working on getting the multi-touch gestures working with this build. I switched to Voodool2cController and it was working great until I upgraded to Mojave 10.14.5, then it just stopped working. I found a new revised VoodooPS2Controller kext from 
[acidanthera](https://github.com/acidanthera/VoodooPS2), and it works beautifully. All the gestures work and so does force click. The last thing that I noticed is that tap to drag now works, which it did not when enabled with VoodooI2C Hope this helps .   
[EFI.zip](https://github.com/bavariancake/XPS9570-macOS/files/3219613/EFI.zip)

**Comment by kevinshroff** [comment]
> Hi All, I have been working on getting the multi-touch gestures working with this build. I switched to Voodool2cController and it was working great until I upgraded to Mojave 10.14.5, then it just stopped working. I found a new revised VoodooPS2Controller kext from
> [acidanthera](https://github.com/acidanthera/VoodooPS2), and it works beautifully. All the gestures work and so does force click. The last thing that I noticed is that tap to drag now works, which it did not when enabled with VoodooI2C Hope this helps .
> [EFI.zip](https://github.com/bavariancake/XPS9570-macOS/files/3219613/EFI.zip)

That kext makes it feel like all gestures are at like 30 fps, weird feeling (difficult to explain) and not as smooth as voodooi2c.

**Comment by d3myz** [comment]
> > Hi All, I have been working on getting the multi-touch gestures working with this build. I switched to Voodool2cController and it was working great until I upgraded to Mojave 10.14.5, then it just stopped working. I found a new revised VoodooPS2Controller kext from
> > [acidanthera](https://github.com/acidanthera/VoodooPS2), and it works beautifully. All the gestures work and so does force click. The last thing that I noticed is that tap to drag now works, which it did not when enabled with VoodooI2C Hope this helps .
> > [EFI.zip](https://github.com/bavariancake/XPS9570-macOS/files/3219613/EFI.zip)
> 
> That kext makes it feel like all gestures are at like 30 fps, weird feeling (difficult to explain) and not as smooth as voodooi2c, atleast for me

Hmm. that is weird. I switched to using an apple magic trackpad and it's been a game changer. Still using the same kext but the gestures are the same speed as on the trackpad and on my wife's Macbook Air. At first they felt slow to me so I started using the maxmem string that upped my Ram to 2GB, I feel that helped with the gesture animation when switching desktops. I used voodooi2c it at first and I liked it better, but it would cause my trackpad to stop working after a while and the intermittent boot issues were a pain. Now in comparison, the Trackpad makes it feel like being on a realy MBP. Even the voodooi2c pails in comparison to using apple's trackpad, i'd highly recommend picking one up. I found a Gen 1 for $30 and I got lucky and found a used Gen 2 for $40.
